dialect/sql/schema/mysql.go

```diff
@@ -951,5 +951,22 @@ func (d *MySQL) atIndex(idx1 *Index, t2 *schema.Table, idx2 *schema.Index) error
 		}
 		idx2.AddParts(part)
 	}
+	if t, ok := indexType(idx1, dialect.MySQL); ok {
+		idx2.AddAttrs(&mysql.IndexType{T: t})
+	}
 	return nil
 }
+
+func indexType(idx *Index, d string) (string, bool) {
+	ant := idx.Annotation
+	if ant == nil {
+		return "", false
+	}
+	if ant.Type != "" {
+		return ant.Type, true
+	}
+	if ant.Types != nil && ant.Types[d] != "" {
```

Comment:
I'd expect `Types` to take precedence. Understood `Type` as default. And `Types` to override eg SQLite

Reply:
done